--- rc.112/MutableHashMap.ts
+++ 4.0.1/MutableHashMap.ts
@@ -20,7 +20,7 @@
 import { pipeArguments } from "./Pipeable.ts"
 import { hasProperty } from "./Predicate.ts"
 
-const TypeId = "~effect/collections/MutableHashMap"
+const TypeId = "~effect/MutableHashMap"
 
 /**
  * A mutable hash map that stores key-value pairs and supports both referential
@@ -397,10 +397,6 @@
   } else if (isSimpleKey(key)) {
     return Option.none()
   }
-  const refKey = referentialKeysCache.get(self)
-  if (refKey !== undefined) {
-    return self.backing.has(refKey) ? Option.some(self.backing.get(refKey)!) : Option.none()
-  }
   const hash = Hash.hash(key)
   const bucket = self.buckets.get(hash)
   if (bucket === undefined) {
@@ -409,7 +405,6 @@
   return getFromBucket(self, bucket, key)
 })
 
-const referentialKeysCache = new WeakMap<any, any>()
 const isSimpleKey = (u: unknown): boolean => typeof u !== "object" && typeof u !== "function"
 
 /**
@@ -483,7 +478,6 @@
   for (let i = 0, len = bucket.length; i < len; i++) {
     if (Equal.equals(key, bucket[i])) {
       const refKey = bucket[i]
-      referentialKeysCache.set(key, refKey)
       return Option.some(self.backing.get(refKey)!)
     }
   }
@@ -839,12 +833,6 @@
     self.backing.set(key, value)
     return self
   }
-  let refKey = referentialKeysCache.get(self)
-  if (refKey !== undefined && self.backing.has(refKey)) {
-    self.backing.set(refKey, value)
-    return self
-  }
-
   const hash = Hash.hash(key)
   const bucket = self.buckets.get(hash)
   if (bucket === undefined) {
@@ -853,7 +841,7 @@
     return self
   }
 
-  refKey = getRefKey(bucket, key)
+  let refKey = getRefKey(bucket, key)
   if (refKey === undefined) {
     bucket.push(key)
     refKey = key
@@ -868,7 +856,6 @@
 ) => {
   for (let i = 0, len = bucket.length; i < len; i++) {
     if (Equal.equals(key, bucket[i])) {
-      referentialKeysCache.set(key, bucket[i])
       return bucket[i]
     }
   }
@@ -1083,19 +1070,13 @@
     }
     return self
   }
-  let refKey = referentialKeysCache.get(self)
-  if (refKey !== undefined && self.backing.has(refKey)) {
-    self.backing.set(refKey, f(self.backing.get(refKey)!))
-    return self
-  }
-
   const hash = Hash.hash(key)
   const bucket = self.buckets.get(hash)
   if (bucket === undefined) {
     return self
   }
 
-  refKey = getRefKey(bucket, key)
+  const refKey = getRefKey(bucket, key)
   if (refKey === undefined) {
     return self
   }
@@ -1607,15 +1588,14 @@
     return self
   }
 
-  const key = referentialKeysCache.get(self) ?? key_
-  const hash = Hash.hash(key)
+  const hash = Hash.hash(key_)
   const bucket = self.buckets.get(hash)
   if (bucket === undefined) {
     return self
   }
   for (let i = 0, len = bucket.length; i < len; i++) {
     const bkey = bucket[i]
-    if (bkey === key || Equal.equals(key, bkey)) {
+    if (bkey === key_ || Equal.equals(key_, bkey)) {
       self.backing.delete(bkey)
       bucket.splice(i, 1)
       break
